tests/integration/test_task_client.py

```diff
@@ -0,0 +1,122 @@
+import time
+from typing import Generator
+
+import pytest
+
+from pyatlan.client.atlan import AtlanClient
+from pyatlan.model.assets import Column
+from pyatlan.model.enums import (
+    AtlanConnectorType,
+    AtlanTagColor,
+    AtlanTaskStatus,
+    AtlanTaskType,
+    SortOrder,
+)
+from pyatlan.model.fluent_tasks import FluentTasks
+from pyatlan.model.search import SortItem
+from pyatlan.model.task import AtlanTask, TaskSearchRequest
+from pyatlan.model.typedef import AtlanTagDef
+from tests.integration.client import TestId
+
+MODULE_NAME = TestId.make_unique("TaskClient")
+TAG_NAME = MODULE_NAME
+
+DB_NAME = "RAW"
+TABLE_NAME = "PACKAGETYPES"
+COLUMN_NAME = "PACKAGETYPENAME"
+SCHEMA_NAME = "WIDEWORLDIMPORTERS_WAREHOUSE"
+
+
+@pytest.fixture(scope="module")
+def snowflake_conn(client: AtlanClient):
+    return client.asset.find_connections_by_name(
+        "development", AtlanConnectorType.SNOWFLAKE
+    )[0]
+
+
+@pytest.fixture(scope="module")
+def snowflake_column_qn(snowflake_conn):
+    return f"{snowflake_conn.qualified_name}/{DB_NAME}/{SCHEMA_NAME}/{TABLE_NAME}/{COLUMN_NAME}"
+
+
+@pytest.fixture()
+def snowflake_column(client: AtlanClient, snowflake_column_qn) -> Column:
+    client.asset.add_atlan_tags(
+        asset_type=Column,
+        qualified_name=snowflake_column_qn,
+        atlan_tag_names=[TAG_NAME],
+        propagate=True,
+        remove_propagation_on_delete=True,
+        restrict_lineage_propagation=True,
+    )
+    snowflake_column = client.asset.get_by_qualified_name(
+        snowflake_column_qn, asset_type=Column
+    )
+    return snowflake_column
+
+
+@pytest.fixture()
+def task_search_request(snowflake_column: Column) -> TaskSearchRequest:
+    return (
+        FluentTasks()
+        .page_size(1)
+        .sort(
+            by=SortItem(
+                field=AtlanTask.START_TIME.numeric_field_name,
+                order=SortOrder.DESCENDING,
+            )
+        )
+        .where(AtlanTask.ENTITY_GUID.eq(snowflake_column.guid))
+        .where(AtlanTask.TYPE.eq(AtlanTaskType.CLASSIFICATION_PROPAGATION_ADD.value))
+        .where(AtlanTask.STATUS.match(AtlanTaskStatus.PENDING.value))
```

Comment:
Let's just keep an eye on this — my only concern here is that if we filter by `PENDING` there's a chance that the background task is actually picked up and run (and then no longer pending) before this test runs...

For repeatability, it might be safer to remove this pending condition — searching by entity GUID (will always be unique) alone may be sufficient, but GUID + the task type (classification_propagation_add) should be sufficient, I think?

Reply:
Right! 
I've modified the search request to utilize the entity GUID along with the task type 👍